### PR TITLE
Deliverers' ID attribute recognised by Specberus, and spat out as part of metadata

### PR DESCRIPTION
**Pull request by tripu**
We haven't reached consensus in #130. But @deniak and I favoured reading a new attribute `data-deliverer-id` on documents, and it's the quickest solution for the issue. So here is that solution.

For more info about how those IDs are being returned now, see [the updated documentation](https://github.com/w3c/specberus/blob/tripu/issue-130-deliverers-metadata/README.md#emitting-metadata-about-the-document).

Also, notice that the new attribute **is not required** by Specberus, ie not finding it on the document will *not* throw an error.

### CODE REVIEW

**Comment by astorije** [comment]
@deniak, @tripu and I discussed this on the phone today. We will use @dontcallmedom's suggestion to extract the WG ID from the PP link, to get to the point faster, but some of us express concerns about this solution. @plh had ideas and we will most likely refactor after v1.

@tripu, I will review when you are done, no rush :)

**Comment by tripu** [comment]
@astorije: change done; please review.

File: README.md [hunk 1]
```diff
@@ -84,11 +84,14 @@ These properties are now returned when found:
...
 * `process`: The process rules, **as they appear on the text of the document**, eg `'14 October 2005'`.
-* `deliverers`: The deliverer(s) responsible for the document (WGs, TFs, etc).
+* `deliverers`: The deliverer(s) responsible for the document (WGs, TFs, etc); an `Array` of `Object`s, each one with these properties:
+  * `homepage`: URL of the grop's home page.
+  * `name`: name of the group, exactly as it is found in the hyperlink on the document.
 * `thisVersion`: URL of this version of the document.
 * `previousVersion`: URL of the previous version of the document (the last one, if multiple are shown).
 * `latestVersion`: URL of the latest version of the document.
-* `editorIDs`: ID(s) of the editor(s) responsible for the document, necessarily matching the pattern `/\d+/`.
+* `editorIDs`: ID(s) of the editor(s) responsible for the document (an `Array` of `String`s, necessarily matching the pattern `/\d+/`).
```
Comment (astorije) [inline README.md]: Why do you need to specify that this is an `Array` of `String`s? Isn't it possible to send an array of integers instead?

File: README.md [hunk 2]
```diff
@@ -84,11 +84,14 @@ These properties are now returned when found:
...
+* `delivererID` ID of the group, (a `String`, necessarily matching the pattern `/\d+/`).
```
Comment (astorije) [inline README.md]: Why isn't this part of the `deliverers`? Something like:
- `deliverers`
  - `id`
  - `homepage`
  - `name`

Also, there can be multiple deliverers, so there will be multiple IDs detected.

File: lib/rules/heuristic/group.js [hunk 3]
```diff
@@ -4,16 +4,28 @@
...
 exports.check = function (sr, done) {
-    var patterns = /.+ interest group$|.+ community group$|.+ working group$/i;
-    var candidates = [];
-    var item;
-    var i;
+    var patterns = /.+ interest group$|.+ community group$|.+ working group$/i
+    ,   candidates = []
+    ,   candidate
+    ,   item
+    ,   i;
```
Comment (astorije) [inline lib/rules/heuristic/group.js]: This is such an ugly, error-prone syntax! (I am not blaming you but the very first guy who used it :) )

FYI, I plan to switch them back all to the traditional `var [...];` one. This horror has to stop and the world needs a leader :D

File: lib/rules/sotd/pp.js [hunk 4]
```diff
@@ -64,17 +70,25 @@ exports.check = function (sr, done) {
...
-        if (/^http:\/\/www\.w3\.org\/2004\/01\/pp-impl\/\d+\/status$/.test(href) &&
-            text === "public list of any patent disclosures" &&
-            $a.attr("rel") === "disclosure") {
-            foundPublicList = true;
-            return;
+        if (/^http:\/\/www\.w3\.org\/2004\/01\/pp-impl\/\d+\/status(#.*)?$/.test(href) &&
+            text === "public list of any patent disclosures") {
+            ids = DELIVERER_ID_REGEX.exec(href);
+            console.log (href + ' - ' + DELIVERER_ID_REGEX);
+            console.dir(ids);
```
Comment (astorije) [inline lib/rules/sotd/pp.js]: Aren't these 2 lines remains from your development tests? If not, how useful is it in general? This is not sarcasm this time, I am genuinely wondering :)

**Comment by astorije** [comment]
@tripu, review done. Mostly, a bit puzzled about the `delivererID` scalar param versus the `deliverers` collection param.

**Comment by tripu** [comment]
@astorije, [I've committed a few changes](https://github.com/w3c/specberus/commit/09c4cafd06f001d89a729f61e9efcf3e2c862c8e), in response to your comments.
Details:

* `delivererIDs` and `editorIDs` are now `Array`s of `Number`s (not `String`s like before).
* Now, multiple `delivererIDs` are returned (it's a vector instead of an scalar). Those IDs can't be part of the `Object`s inside `deliverers` because those two pieces of information are extracted in different ways, from different parts of the document. There's no (consistent) way to associated IDs to names and homepages: there could be a different number of IDs and homepages, and even if the number is the same, how do we know which is which&hellip;?
* Removed `console.log()` and `console.dir()` statements (oops! thanks!).

Additional fixes:

* `README.md`: updated with latest changes to metadata fields, fixed hierarchy of document profiles, fixed typo, improved styling.
* Profile `TR`: fixed a bug (rule `sotd/pp` was being checked twice).

**Comment by astorije** [comment]
Looks good, I just suggest a fix in #135.

**Pull request merged**